### ISSUE TITLE
Order State Transition Emails DO NOT include Email Template Header / Footer of the SalesChannel

### ISSUE THREAD

**Issue by AlexBachmann** [shopware/shopware#654]
**Description**  
If the order / payment / delivery state of an order changes, an email is triggered and send to the customer. Even though the order is related to a specific sales channel, the mail template header / footer of that sales channel is not being rendered for this email.

**Possible Solution**  
The main (but not the only) issue is this block of code in Shopware\Core\Content\MailTemplate\Service\MailService (lines 105 ff.)
```php
$salesChannelId = $data['salesChannelId'];
$salesChannel = null;

if ($salesChannelId !== null && !isset($templateData['salesChannel'])) {
    $criteria = new Criteria([$salesChannelId]);
    $criteria->addAssociation('mailHeaderFooter');
    /** @var SalesChannelEntity|null $salesChannel */
    $salesChannel = $this->salesChannelRepository->search($criteria, $context)->get($salesChannelId);

    if ($salesChannel === null) {
        throw new SalesChannelNotFoundException($salesChannelId);
    }

    $templateData['salesChannel'] = $salesChannel;
}
```
This code sets the $salesChannel variable to NULL. This $salesChannel variable will later be responsible to render the MailTemplate Header & Footer for the email, so it is essential, that it is set correctly. However, this is not the case. 

The Shopware\Core\Checkout\Order\Api\OrderActionController provides the salesChannelId in the $data bag provided to the method, but it also provides the salesChannel data in the $templateData. Therefore, the entire code block is being skipped over. the $salesChannel variables stays NULL and no header footer template is being rendered. 

**The bigger issue with this implementation**

At the beginning I stated, that this code block is the main issue, but not the only issue with this implementation. There is a bigger issue at play here, which is: **the calling function needs to know about the inner workings of this method** for the template to be rendered correctly.

What do I mean by that?

One could think that this issue could be easily fixed by replacing the following code
```php
$salesChannelId = $data['salesChannelId'];
$salesChannel = null;

if ($salesChannelId !== null && !isset($templateData['salesChannel'])) {
    ....
}
...
```
with this code
```php
$salesChannelId = $data['salesChannelId'];
$salesChannel = null;

if ($salesChannelId !== null && !isset($templateData['salesChannel'])) {
    ....
} else if (isset($templateData['salesChannel'])) {
   $salesChannel = $templateData['salesChannel'];
}
```
However, this wouldn't fix the issue either, because it is very possible to pass the $salesChannel data to the $templateData, without resolving $mailHeaderFooter property of the SalesChannel object. (This is exactly what happens in the OrderActionController by the way. The salesChannel object is passed from the order entity to the $templateData array, however, the order entity does not resolve the mailHeaderFooter property of its salesChannel entity.).

The mailService should be resposible to load the required header / footer data on it's own, if it is not yet loaded via the salesChannel object. It cannot trust that the calling function knows, that this data is required for the method to work properly.

**Comment by pantrtxp** [comment]
HI @AlexBachmann,

this issue is known to us and will probably be fixed with version 6.2.1
you can keep track of the status of this issue [here](https://issues.shopware.com/issues/NEXT-6430)

Sunny regards
Pantrtxp

**Comment by pantrtxp** [comment]
I am closing this Issue due to inactivity and being duplicate.

You can track the state of our progress with the given link at any time 
@AlexBachmann, thank you for your concern :blue_heart: 

Sunny regards
Pantrtxp
